Elevel/Application/Features/ReportCommands/CreateReportCommand.cs

```diff
@@ -82,6 +84,46 @@ public async Task<Response> Handle(Request request, CancellationToken cancellati
             {
                 var report = _mapper.Map<Report>(request);
                 
+                if (request.AuditionId == null && request.QuestionId != null)
+                {
+                    var question = await _context.Questions
+                        .AsNoTracking()
+                        .FirstOrDefaultAsync(x => x.Id == request.QuestionId, cancellationToken: cancellationToken);
+                    
+                    if (question == null)
+                    {
+                        throw new NotFoundException(nameof(question), request.QuestionId);
+                    }
+
+                    report.CreatorId = question.CreatorId;
+                }
+                else if(request.AuditionId != null && request.QuestionId != null)
+                {
+                    var audition = await _context.Auditions
+                        .AsNoTracking()
+                        .FirstOrDefaultAsync(x => x.Id == request.AuditionId, cancellationToken: cancellationToken);
+
+                    if (audition == null)
+                    {
+                        throw new NotFoundException(nameof(audition), request.AuditionId);
+                    }
+
+                    report.CreatorId = audition.CreatorId;
+                }
+                else if(request.TopicId != null)
```

Comment:
What if topicId != null and QuestionId != null?

Reply:
Or eerything != null?

Reply:
It is not possible because validator will block this case